src/Clock.tsx

```diff
@@ -0,0 +1,60 @@
+import { Component } from 'react';
+
+type Props = {
+  clockName: string,
+};
+
+type State = {
+  date: Date,
+};
+
+export class Clock extends Component<Props, State> {
+  state = {
+    date: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      const currentDate = new Date();
+
+      this.setState({ date: currentDate });
+      // eslint-disable-next-line
+      console.log(currentDate.toLocaleTimeString());
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const prevName = prevProps.clockName;
+    const currName = this.props.clockName;
+
+    if (prevName !== currName) {
+      // eslint-disable-next-line
+      console.log(`Renamed from ${prevName} to ${currName}`);
+    }
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+  }
+
+  render() {
+    const { clockName } = this.props;
+    const { date } = this.state;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {clockName}
+        </strong>
+
+        time is
+
```

Comment:
Spaces are needed before and after 'time is'

Reply:
got it